New Script: Prometheus Alertmanager

andygrunwald opened the pull request:
## ✍️ Description

This Pull Request provides a new Script: Prometheus Alertmanager.

This is the Alerting-Part of Prometheus, see https://prometheus.io/docs/alerting/latest/overview/

Prometheus is already part of this repository.

- - -
- Related Discussion: https://github.com/community-scripts/ProxmoxVE/discussions/1005

---

## 🛠️ Type of Change
Please check the relevant options:  
- [ ] Bug fix (non-breaking change that resolves an issue)  
- [ ] New feature (non-breaking change that adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [X] New script (a fully functional and thoroughly tested script or set of scripts)  

---

## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [X] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [X] Testing performed (I have thoroughly tested my changes and verified expected functionality.)  
- [X] Documentation updated (I have updated any relevant documentation)

---

## 📋 Additional Information (optional)

This is a near copy of the Prometheus scripts.
The setup is 1:1 the same.

I would appreciate a proper testing from your side as well.
This is my first script contribution.

andygrunwald committed:
New Script: Prometheus Alertmanager

MickLesk commented:
Why not integrating in existing Prometheus Script als optional Installation/addon?

andygrunwald commented:
@MickLesk This would be an option. I didn't do this for two reasons:

1. The standalone implementation was very straightforward
2. The tooling around this ecosystem (Prometheus, Alertmanager, VictoriaMetrics with vmalert, vmagent or Thanos) is component-based by design with the usecase to "mix and match" like you want - This means there are usecases where you just want to have an Alertmanager without Prometheus (e.g., just to use the Alerting system, but not the metrics system)

Re 1): Is there an example script where I can get some orientation about how to do it?

Re 2): There might be an argument that those use cases would not use the Proxmox scripts, and this might be right. For a home lab to experiment, this may still be valid.

---

Should this stay standalone or should this be integrated into the Prometheus script?
My personal preference would be the standalone version, also due to maintenance.

MickLesk commented on ct/prometheus-alertmanager.sh:
```diff
@@ -0,0 +1,68 @@
+
+# App Default Values
+APP="Prometheus-Alertmanager"
+var_tags="monitoring,alerting"
```
;

andygrunwald replied:
What do you mean by this? Is an `;` expected at the end of line 10?
I was not able to see this in other scripts.

dsiebel replied:
`;` is used to separate multiple tags, not `,`.

andygrunwald replied:
Thx @dsiebel 

Fixed in https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/d59aed09dead5878c83284c04eb88c0458e59ba5

MickLesk commented on ct/prometheus-alertmanager.sh:
```diff
@@ -0,0 +1,68 @@
+    systemctl stop prometheus-alertmanager
+    msg_ok "Stopped ${APP}"
+
+    msg_info "Updating ${APP} to ${RELEASE}"
```
v${RELEASE}

andygrunwald replied:
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/958aa957bb9d1b24ad94b4697d0ed32a052121e5

MickLesk commented on ct/prometheus-alertmanager.sh:
```diff
@@ -0,0 +1,68 @@
+    msg_ok "Stopped ${APP}"
+
+    msg_info "Updating ${APP} to ${RELEASE}"
+    wget -q https://github.com/prometheus/alertmanager/releases/download/v${RELEASE}/alertmanager-${RELEASE}.linux-amd64.tar.gz
```
cd /opt before

andygrunwald replied:
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/bc74be10ca9149bbabaf5925056d57699d6972c8

andygrunwald committed:
Alertmanager: Append v before Release-Version

andygrunwald committed:
Alertmanager: Switch to `/opt` before updating release

andygrunwald committed:
Alertmanager: Split `var_tags` by `;`

michelroegl-brunner commented on ct/prometheus-alertmanager.sh:
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2025 tteck
```
Old copyright

andygrunwald replied:
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/07000662330ce13f7f23645c952a8e753ff0395b

andygrunwald committed:
Alertmanager: Set correct copyright

Commit:
Alertmanager: Change installation of curl, sudo and mc to a oneliner

Co-authored-by: Michel Roegl-Brunner <73236783+michelroegl-brunner@users.noreply.github.com>

andygrunwald commented on install/prometheus-alertmanager-install.sh:
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021-2025 tteck
```
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/07000662330ce13f7f23645c952a8e753ff0395b

Commit:
Alertmanager: Change the way how the systemd unit file is written to disk

Co-authored-by: Michel Roegl-Brunner <73236783+michelroegl-brunner@users.noreply.github.com>

andygrunwald committed:
Alertmanager: Changed the way on how we move and delete files

andygrunwald commented on install/prometheus-alertmanager-install.sh:
```diff
@@ -0,0 +1,62 @@
+mkdir -p /var/lib/alertmanager
+wget -q https://github.com/prometheus/alertmanager/releases/download/v${RELEASE}/alertmanager-${RELEASE}.linux-amd64.tar.gz
+tar -xf alertmanager-${RELEASE}.linux-amd64.tar.gz
+cd alertmanager-${RELEASE}.linux-amd64
```
This is the way how the Prometheus Script is doing it.
However, I changed it now, see https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/1bff355e1d22e3a6cf43f929bc424cda9f6f84eb

andygrunwald commented on install/prometheus-alertmanager-install.sh:
```diff
@@ -0,0 +1,62 @@
+msg_info "Cleaning up"
+$STD apt-get -y autoremove
+$STD apt-get -y autoclean
+rm -rf ../alertmanager-${RELEASE}.linux-amd64 ../alertmanager-${RELEASE}.linux-amd64.tar.gz
```
We are now (see https://github.com/community-scripts/ProxmoxVE/pull/1272/commits/1bff355e1d22e3a6cf43f929bc424cda9f6f84eb) in the same folder. Hence, I removed the `../`

andygrunwald committed:
Alertmanager: Make files executable

andygrunwald commented:
As far as I can see, the license is there, but the "Validate scripts" Github action still fails.

andygrunwald committed:
Alertmanager: Set correct license and source header

andygrunwald commented:
@michelroegl-brunner @MickLesk I rebased the branch against latest `main` and fixed all CI checks.

The CI check "Validate scripts / Check changed files (pull_request) " is failing, due to
```
data: {
      message: 'Resource not accessible by integration',
      documentation_url: 'https://docs.github.com/rest/issues/comments#create-an-issue-comment',
      status: '403'
    }
```

This seem to be unrelated to the change and rather related to the CI check itself.

MickLesk committed:
Update prometheus-alertmanager.json
